test/csuite/wt8246_compact_rts_data_correctness/main.c

```diff
@@ -0,0 +1,378 @@
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+#include <sys/wait.h>
+#include <signal.h>
+
+#define TIMEOUT 4
+
+#define NUM_RECORDS 1200000
+
+#define ENV_CONFIG_REC "log=(archive=false,recover=on)"
+/* Constants and variables declaration. */
+/*
+ * You may want to add "verbose=[compact,compact_progress]" to the connection config string to get
+ * better view on what is happening.
+ */
+static const char conn_config[] =
+  "create,cache_size=1GB,statistics=(all),statistics_log=(wait=1,json=true,on_close=true)";
+static const char table_config_row[] =
+  "allocation_size=4KB,leaf_page_max=4KB,key_format=Q,value_format=QQQS";
+static const char table_config_col[] =
+  "allocation_size=4KB,leaf_page_max=4KB,key_format=r,value_format=QQQS";
+static char data_str[1024] = "";
+static const char working_dir_row[] = "WT_TEST.compact-stress-row";
+static const char working_dir_col[] = "WT_TEST.compact-stress-col";
+static const char compact_file_fmt[] = "%s/compact_started";
+static char value_a[] = "AA";
+static char value_b[] = "BB";
+static char value_c[] = "CC";
+static char value_d[] = "DD";
+
+static void sig_handler(int) WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+
+/* Forward declarations. */
+static int run_test(bool column_store, const char *uri);
+static void populate(WT_SESSION *session, const char *uri);
+static void workload_compact(const char *, const char *, const char *uri);
+static void remove_records(WT_SESSION *session, const char *uri, int commit_ts);
+static void run_compact(WT_SESSION *session, const char *uri);
+static void large_updates(WT_SESSION *session, const char *uri, char *value, int commit_ts);
+static void check(WT_SESSION *session, const char *uri, char *value, int commit_ts);
+
+/*
+ * Signal handler to catch if the child died unexpectedly.
+ */
+static void
+sig_handler(int sig)
+{
+    pid_t pid;
+
+    WT_UNUSED(sig);
+    pid = wait(NULL);
+    /*
+     * The core file will indicate why the child exited. Choose EINVAL here.
+     */
+    testutil_die(EINVAL, "Child process %" PRIu64 " abnormally exited", (uint64_t)pid);
+}
+
+/* Methods implementation. */
+int
+main(int argc, char *argv[])
+{
+    TEST_OPTS *opts, _opts;
+
+    opts = &_opts;
+    memset(opts, 0, sizeof(*opts));
+    testutil_check(testutil_parse_opts(argc, argv, opts));
+
+    testutil_assert(run_test(false, opts->uri) == EXIT_SUCCESS);
+
+    testutil_assert(run_test(true, opts->uri) == EXIT_SUCCESS);
+
+    testutil_cleanup(opts);
+
+    return (EXIT_SUCCESS);
+}
+
+static void
+run_compact(WT_SESSION *session, const char *uri)
+{
+    printf("Compact start...\n");
+    /* Perform compact operation. */
+    testutil_check(session->compact(session, uri, NULL));
+    printf("Compact end...\n");
```

Comment:
Ideally we want the child process to be killed before this message gets printed. Have you confirmed this?

Reply:
@hbokhari Yes, I ran multiple times in my local machine and confirmed that the compact is killed before printing `Compact end...`. It prints `Compact start...` and `Compact process interrupted and killed...`

Reply:
I guess it depends on the hardware the test is running on and on a general load of the system. Technically there's a possibility that the parent process gets starved to the point where compact has finished, or compact finishes faster than 4 sec. But if that happens we'll see `Child finished processing...` message and the child process will fail with `EXIT_FAILURE`.